### DISCUSSION TITLE
.env/secrets file loader should be user injectable (for pydantic 2?)

### DISCUSSION THREAD

**Discussion by 07pepa** [pydantic/pydantic#4481] [Feature Request]
currently env file loader can not be swaped to some custom one 

https://github.com/pydantic/pydantic/blob/eccd85e4d012e70ffbd81f379179da900d4621c5/pydantic/env_settings.py#L212

that is to allow loading enviroment file from other place ( aws s3, etc)

Because particulary on aws it is better to put enviroment files on s3 then on fs (that is not shared acros multiple resources)

imho only thing that should be injectable should be provider of byte array or bytes or byte stream that will recive path

and can eather return or throw some standardized  (pydantic) exeption like path (does not exist, is not file, not enought premision to read it, etc)

same can be done for secrets (and aws secret manager)
Thanks for considering this

**Comment by samuelcolvin** [comment]
Totally agree, parsing env variables should be a [Settings Source](https://pydantic-docs.helpmanual.io/usage/settings/#customise-settings-sources) like other things, it should be use by default, but easily not used/switched off.
